tokio-quiche/src/quic/mod.rs

```diff
@@ -187,25 +208,27 @@ where
 /// Sharing a socket among multiple connections will lead to lost packets as
 /// both connections try to read from the shared socket.
 pub async fn connect_with_config<Tx, Rx, S, App>(
-    socket: S, host: Option<&str>, params: &ConnectionParams<'_>, app: App,
+    sockets: S, host: Option<&str>, params: &ConnectionParams<'_>, app: App,
 ) -> QuicResult<QuicConnection>
 where
-    Tx: DatagramSocketSend + Send + 'static,
-    Rx: DatagramSocketRecv + Unpin + 'static,
-    S: TryInto<Socket<Tx, Rx>>,
+    Tx: DatagramSocketSend + Send + Clone + 'static,
+    Rx: DatagramSocketRecv + Clone + Unpin + 'static,
+    S: TryInto<MultiSocket<Tx, Rx>>,
     S::Error: std::error::Error + Send + Sync + 'static,
     App: ApplicationOverQuic,
 {
-    let socket: Socket<Tx, Rx> = socket.try_into()?;
+    let sockets: MultiSocket<Tx, Rx> = sockets.try_into()?;
+    let mut caps = SocketCapabilities::default();
 
     #[cfg_attr(not(target_os = "linux"), expect(unused_mut))]
-    let mut caps = SocketCapabilities::default();
-    #[cfg(target_os = "linux")]
-    if let Some(s) = socket.as_udp_socket() {
-        caps = SocketCapabilities::apply_all_and_get_compatibility(
-            s,
-            params.settings.max_send_udp_payload_size,
-        );
+    for socket in sockets.paths.iter() {
+        #[cfg(target_os = "linux")]
```

Comment:
Similarly, these operations only make sense in a Linux target; no need for loops in other ones.
```suggestion
    #[cfg(target_os = "linux")]
    for socket in sockets.paths.iter() {
```